src/types/types.ts

```diff
@@ -0,0 +1,37 @@
+import {Path} from 'react-router-dom';
+
+export interface Location<T> extends Path {
+    state: T;
+}
+
```

Comment:
besides the `Location` interface, that needs to extend from `Path` interface,  all other can be **type** , which I can add `Type` in the end of each type, preventing naming issues with the components or variables